src/builder.rs

```diff
@@ -391,6 +396,21 @@ fn prepare_header(dst: &mut Write, header: &mut Header, path: &Path) -> io::Resu
     Ok(())
 }
 
+fn prepare_header_link(dst: &mut Write, header: &mut Header, link_name: &Path) -> io::Result<()> {
+    // Same as previous function but for linkname
+    if let Err(e) = header.set_link_name(&link_name) {
+        let data = path2bytes(&link_name)?;
+        if data.len() < header.as_old().linkname.len() {
```

Comment:
Man really all I have is comments on the old code... So along those lines I actually have no idea what this is checking, but it seems like it's trying to check for errors not related to length, but for that I think it might be more appropriate to inspect the `e` error we got instead of checking lengths afterwards

Reply:
It's verifying that `set_link_name` failed because of the length being too long, if not, return `e`. We get `ErrorKind::Other` (from `copy_into` in src/header.rs)  if the name is too long, which isn't specific enough to differentiate from other errors unfortunately.

Reply:
Ok yeah that's what I figured. Also agreed there's not a great way to distinguish errors for now, but mind leaving a comment indicating what this pattern is doing?

Reply:
Sure